### PR TITLE
Address UIDelays in AbstractGraphProvider.GetCommands

### PR DESCRIPTION
**Pull request by ToddGrun**
Both VB and C# derivations of this class were getting called for the same items (when both types of projects were active in a solution). As the implementations only differed based on project capability supported, instead just export the base class with a project capability for both C# and VB. This should reduce the effective cost of this method in half.

Partially addresses [AB#1936654](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1936654)

### CODE REVIEW

**Comment by ToddGrun** [comment]
The previous attempt at optimizing the GetCommands method directly was [here](https://github.com/dotnet/roslyn/pull/71540). That approach added complexity and had functional issues.

**Comment by jasonmalinowski** [comment]
> Both VB and C# derivations of this class were getting called for the same items (when both types of projects were active in a solution).

This feels like a bug in progression: I'd expect the handlers who are responding to that particular project are called, not any random provider. For example, are we also being called for stuff produced by C++? That said, if we can make things faster by deleting code, I'm all for that!

File: src/VisualStudio/Core/Def/Progression/GraphProvider.cs [hunk 1]
```diff
@@ -2,16 +2,16 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Progression/GraphProvider.cs:5]: Nice, thanks!

**Review by jasonmalinowski — APPROVED** [review]

**Comment by ToddGrun** [comment]
I brought this up with them and they weren't particularly keen on changing the behavior at this point. I've verified that the merged handler is called for both VB and C# projects and not for C++ projects.

---
In reply to: [1899462667](https://github.com/dotnet/roslyn/pull/71707#issuecomment-1899462667) [](http://example.com/codeflow?ancestors=1899462667)

**Pull request merged**